### PR TITLE
[Tensorflow][Test]Telemetry test for Tensorflow Inference

### PR DESCRIPTION
**Pull request by TusharKanekiDey**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [build] | [test] | [build, test] | [ec2, ecs, eks, sagemaker]

*Description:*
Adding telemetry test for TF Inference

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py [hunk 1]
```diff
@@ -116,3 +130,12 @@ def host_setup_for_tensorflow_inference(serving_folder_path, framework_version,
...
         ec2_connection.run(f"cp -r {local_scripts_path} {serving_folder_path}")
+
+
+def check_telemetry(ec2_connection, container_name):
+    ec2_connection.run(
+        f"docker exec -it -user root {container_name}  /bin/bash ",
+        hide=True, warn=True
+    )
+    telemetry_cmd = "python -c 'import os;assert (os.path.exists('/tmp/test_request.txt'))'"
+    ec2_connection.run(telemetry_cmd)
```
Comment (YYStreet) [inline test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py]: try put line140 141 in line 137
```
f"docker exec -it -user root {container_name}  {telemetry_cmd}"
```

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py [hunk 2]
```diff
@@ -45,7 +45,19 @@ def test_ec2_tensorflow_inference_eia_gpu(tensorflow_inference_eia, ec2_connecti
...
+    run_ec2_tensorflow_inference(tensorflow_inference, ec2_connection, "8500", region, True)
+
+
+@pytest.mark.model("mnist")
+@pytest.mark.parametrize("ec2_instance_type", ["c5.4xlarge"], indirect=True)
+def test_ec2_tensorflow_inference_cpu_telemetry(tensorflow_inference, ec2_connection, region, cpu_only):
+    run_ec2_tensorflow_inference(tensorflow_inference, ec2_connection, "8500", region, True)
+
+
+def run_ec2_tensorflow_inference(image_uri, ec2_connection, grpc_port, region, Telemetry_mode=False):
```
Comment (tejaschumbalkar) [inline test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py:60]: nit: `Telemetry_mode` lowercase variable name

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py [hunk 3]
```diff
@@ -77,6 +89,8 @@ def run_ec2_tensorflow_inference(image_uri, ec2_connection, grpc_port, region):
         test_utils.request_tensorflow_inference_grpc(
             script_file_path=mnist_client_path, port=grpc_port, connection=ec2_connection
         )
+        if Telemetry_mode:
```
Comment (tejaschumbalkar) [inline test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py:92]: same here

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py [hunk 4]
```diff
@@ -58,7 +70,7 @@ def run_ec2_tensorflow_inference(image_uri, ec2_connection, grpc_port, region):
     docker_cmd = "nvidia-docker" if "gpu" in image_uri else "docker"
     docker_run_cmd = (
         f"{docker_cmd} run -id --name {container_name} -p {grpc_port}:8500 "
-        f"--mount type=bind,source={model_path},target=/models/mnist -e MODEL_NAME=mnist"
+        f"--mount type=bind,source={model_path},target=/models/mnist -e TEST_MODE=1 -e MODEL_NAME=mnist"
```
Comment (YYStreet) [inline test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py:73]: Has the script been updated to accept this env change?

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py (hunk 2 above)
Comment (YYStreet) [inline test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py:60]: nit: Could it be better to set the first character of it not to capital, a.k.a "telemetry_mode"?

**Comment by TusharKanekiDey** [comment]
> Why do we need test_ec2_tensorflow_inference_cpu_telemetry and test_ec2_tensorflow_inference_cpu in the same time? The first one seems including the second one.

Yes, I think they both are needed and should be separate. Otherwise, it will be difficult to debug what is the exact issue if this test faces any issue.

**Comment by YYStreet** [comment]
> > Why do we need test_ec2_tensorflow_inference_cpu_telemetry and test_ec2_tensorflow_inference_cpu in the same time? The first one seems including the second one.
> 
> Yes, I think they both are needed and should be separate. Otherwise, it will be difficult to debug what is the exact issue if this test faces any issue.

Should we set the first test (telemetry one) dedicated for testing telemetry purpose? In current scenario, 
`test_utils.request_tensorflow_inference_grpc( )` is executed twice in 2 different tests.

Or do you mean `test_utils.request_tensorflow_inference_grpc( )` is a prerequisite to check the existence of `/tmp/test_request.txt`?

**Comment by TusharKanekiDey** [comment]
> > > Why do we need test_ec2_tensorflow_inference_cpu_telemetry and test_ec2_tensorflow_inference_cpu in the same time? The first one seems including the second one.
> > 
> > 
> > Yes, I think they both are needed and should be separate. Otherwise, it will be difficult to debug what is the exact issue if this test faces any issue.
> 
> Should we set the first test (telemetry one) dedicated for testing telemetry purpose? In current scenario,
> `test_utils.request_tensorflow_inference_grpc( )` is executed twice in 2 different tests.
> 
> Or do you mean `test_utils.request_tensorflow_inference_grpc( )` is a prerequisite to check the existence of `/tmp/test_request.txt`?

Yes, we need to run the complete inference process to make sure that tmp file is created.

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py (hunk 4 above)
Comment (TusharKanekiDey) [reply test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py:73]: The script does not needs to be changed.

**Pull request merged**